beginner_source/quickstart/autograd_tutorial.py

```diff
@@ -0,0 +1,290 @@
+# trained the model and just want to apply it to some input data, i.e. we
+# only want to do *forward* computations through the network. We can stop
+# tracking computations by surrounding our computation code with
+# ``torch.no_grad()`` block:
+#
+
+z = torch.matmul(x, w)+b
+print(z.requires_grad)
+
+with torch.no_grad():
+    z = torch.matmul(x, w)+b
+print(z.requires_grad)
+
+
+######################################################################
+# Another way to achieve the same result is to use the ``detach()`` method
+# on the tensor:
+#
+
+z = torch.matmul(x, w)+b
+z_det = z.detach()
+print(z_det.requires_grad)
+
+######################################################################
+# There are reasons you might want to disable gradient tracking:
+#   - To mark some parameters in your neural network at **frozen parameters**. This is
+#     a very common scenario for
+#     `finetuning a pretrained network <https://pytorch.org/tutorials/beginner/finetuning_torchvision_models_tutorial.html>`__
+#   - To **speed up computations** when you are only doing forward pass, because computations on tensors that do
+#     not track gradients would be more efficient.
+
+
+######################################################################
+# Example of Gradient Descent
+# ---------------------------
+#
+# Let's use the AutoGrad functionality to minimize a simple function of
+# two variables :math:`f(x_1,x_2)=(x_1-3)^2+(x_2+2)^2`. We will use the
+# ``x`` tensor to represent the coordinates of a point. To do the gradient
+# descent, we start with some initial value :math:`x^{(0)}=(0,0)`, and
+# compute each consecutive step using:
+#
+# .. math::
+#
+#
+#    x^{(n+1)} = x^{(n)} - \eta\nabla f
+#
+# Here :math:`\eta` is so-called **learning rate** (we will call it ``lr``
+# in our code), and
+# :math:`\nabla f = (\frac{\partial f}{\partial x_1},\frac{\partial f}{\partial x_2})`
+# is the gradient of :math:`f`.
+#
+# We will start by defining the initial value of ``x`` and the function
+# ``f``:
+#
+
+x = torch.zeros(2, requires_grad=True)
+
+def f(x):
+    return (x-torch.tensor([3, -2])).pow(2).sum()
+
+lr = 0.1
+
+
+######################################################################
+# For the gradient descent, let's do 15 iterations. On each iteration, we
+# will update the coordinate tensor ``x`` and print its coordinates to
+# make sure that we are approaching the minimum:
+#
+
+for i in range(15):
+    y = f(x)
+    y.backward()
+    gr = x.grad
+    x.data.add_(-lr*gr)
+    x.grad.zero_()
+    print("Step {}: x[0]={}, x[1]={}".format(i, x[0], x[1]))
+
+
+######################################################################
+# As you can see, we have obtained the values close to the optimal point
+# :math:`(3,-2)`. Training a neural network is in fact a very similar
+# process, we will need to do a number of iterations to minimize the value
+# of the **loss function**.
```

Comment:
We should remove this section for the following reasons:
a) gradient descent is better situated along with optimizers
b) using tensor.data is *highly* discouraged

```suggestion

```